### ISSUE TITLE
Option to keep Participant Count threshold for Tile View

### ISSUE THREAD

**Issue by Freddie-m** [jitsi/jitsi-meet#11166]
The latest unstable version appears to have reverted to setting the participant count threshold for tile view to be triggered to 2. For a few unstable versions, this was set to 3 **AND IT WAS BEAUTIFUL**! 😃  Understandably, some may prefer to keep it at 2, but for those of us who actually like the threshold set to 3, could you please make it a configurable option in config.js? The 3 long tiles was just really beautiful and very functional in meetings where each tile only had one participant.

Thanks for your kind consideration.

**Comment by Freddie-m** [comment]
Confirmed addressed in v7099-1. 
Thanks.

**Comment by Freddie-m** [comment]
Unfortunately, this is not actually addressed. Not sure how I seemed to have thought it was back working again.

**Comment by saghul** [comment]
We are currently working on a number of layout changes, but that has not changed. I just tested on beta.meet.jit.si and it shows the 3 tiles in one row. Do you see otherwise?

**Comment by Freddie-m** [comment]
😳  It's strange. I just tested (on alpha; beta is down - looks like a punctuation error there). So, on one machine, with **Chromium 97** and **Chrome 99**, it's showing me the previous layout with threshold set to 2.

![Screen Shot 2022-03-30 at 2 09 01 AM](https://user-images.githubusercontent.com/63688942/160763016-efd116cf-b2b3-43b8-af76-8a512ec4c1fa.png)

If I add Firefox (95.0.2) as the 3rd participant, I see the new layout with threshold at 3:


![Screen Shot 2022-03-30 at 2 11 18 AM](https://user-images.githubusercontent.com/63688942/160763336-cede9dd2-8211-41ec-a192-be1da3980abe.png)

Interestingly, on **_another_** machine (**Chromium 88**, **Chrome 98**), I see the new layout (without adding a Firefox user).

**Comment by hristoterezov** [comment]
@Freddie-m I've just merged https://github.com/jitsi/jitsi-meet/pull/11226 . 

Could you please test what is the situation on the first machine with this PR deployed?

Note that we don't have a hard limit to display always 3 participants in a single row. So you will see on some screens 3 participants displayed on 2 rows.

After deploying the PR if you still see 3 participants on 2 rows, could you please post the window size? This way I can check if this is supposed to happen and I'll be able to explain why is it happening.

**Comment by Freddie-m** [comment]
@hristoterezov Funny, I was just writing you here. 😃. 
Thanks for the merge. I'll test now and feedback.

**Comment by Freddie-m** [comment]
@hristoterezov Just tested. Unfortunately, it didn't give me the desired behavior. I'm testing on 2 machines:

Machine 1 - MBP, Display Size-17"; Resolution - 1920x1200
Machine 2 - Mac Pro; Display Size - 27"; Resolution - 2560 x 1440

On Machine 1, I see all tiles in one row.
On Machine 2, I see 2 rows (2 tiles in top row and one in bottom row)

I think there's potentially an issue with this. From what I understand, you're saying the number of rows is now tied to the screen resolution/display size. Which means, in the same meeting, 2 different participants on different machines will see something different. The reason this is an issue is that participant seeing one row (wrongly) assumes what they see on screen (limited aspect ratio) is all that's being shown to the other participants, whereas, participants seeing two rows with a 16:9 ratio see more of the participants' streams. _Potentially_ a privacy concern, in my mind. 

Again, note that previously, this worked as preferred - 3 tiles all displayed on the same row regardless of screen size.

**Comment by hristoterezov** [comment]
Hi @Freddie-m,

I'm sorry for the delay! 

I finally tested with 2560 x 1440 window size and I actually see 3 participants in a row. 

In your case, I'm wondering is this the window size or this is just the resolution of the monitor?

Another thing that comes to my mind is that we recently discovered an issue with the recalculations of the layout during resizing the window. Was there some window resizing involved?

> I think there's potentially an issue with this. From what I understand, you're saying the number of rows is now tied to the screen resolution/display size. Which means, in the same meeting, 2 different participants on different machines will see something different. The reason this is an issue is that participant seeing one row (wrongly) assumes what they see on screen (limited aspect ratio) is all that's being shown to the other participants, whereas, participants seeing two rows with a 16:9 ratio see more of the participants' streams. Potentially a privacy concern, in my mind.

If I understand correctly, the main concern here is mostly with using different aspect ratios for the thumbnails and not necessary how many columns and rows do we display. You can actually disable this feature and all thumbnails will be displayed with fixed ratio 16:9 by setting `disableTileEnlargement: true` in config.js.

> Again, note that previously, this worked as preferred - 3 tiles all displayed on the same row regardless of screen size.

I think even before if we didn't have enough width or certain config parameters was enabled we were displaying 2 columns. There are many situations where displaying 3 participants in a row won't be the best way to use the space of the window. For example: if you have a window that the height is 3 times bigger than the width. In this case if we display 3 participants in the row they would take probably 1/3 of the window  and the rest 2/3 will be empty.

**Comment by Freddie-m** [comment]
Hi @hristoterezov,

Thanks for looking into it. 

So, I think I figured out what's going on. First of all, cluing in from your test, I decided to check in fullscreen on my 2560 x 1440 monitor and it indeed displays 3 tiles in a row. However, this is not ideal, because I don't ever run Jitsi in fullscreen mode. So, I exited out of that and tried resizing the window until the width was significantly more than the height and this time, it displayed the 3 tiles in one row. So it appears what you said is what's actually happening - it's factoring in the display area. Although, I have to say, this is a relatively new development - I didn't observe this behavior before (or maybe my window size previously just accommodated 3 tiles in a row).



> If I understand correctly, the main concern here is mostly with using different aspect ratios for the thumbnails and not necessary how many columns and rows do we display.

Yes, the concern is that in essence, viewable remote aspect ratios can be 'modified' by local participants. I suppose one just has to get used to it and remember that the reduced aspect ratio may not be what the other participant sees (this is already the case in a 2-party call anyway if one participant opts for Tile view while the other stays in Stage). In that regard, I don't think there's really anything that can be done besides disabling the behavior entirely like you mentioned.

Thanks again for looking into this. I'll close this out now as it's not a change like I thought it to be. Appreciate your time!
